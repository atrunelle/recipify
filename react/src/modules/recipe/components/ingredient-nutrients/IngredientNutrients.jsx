import React, { Component } from "react";
import recipeService from "../../recipe.service";
import { Chip, Typography, Grid } from "@material-ui/core";

class IngredientNutrients extends Component {
  get allLabels () {
    const dietsLabels = recipeService.getDietLabels(this.props.nutrients.dietLabels);
    const healthLabels = recipeService.getHealthLabels(this.props.nutrients.healthLabels);
    
    return dietsLabels.concat(healthLabels);
  }
  
  get macroNutrientsData () {
    return recipeService.getMacroNutrientsList(this.props.nutrients.totalNutrients);
  }

  render() {
    if (!this.props.nutrients) return false;
    
    const nutrientsView = this.macroNutrientsData.map((nutrient, index) => {
      const quantity = Math.round(nutrient.quantity);

      return (
        <div key={index}>
          <Typography
            gutterBottom
            component="h3">
            { nutrient.label }
          </Typography>
          <Typography
            gutterBottom
            paragraph
            component="p">
              { quantity } { nutrient.unit }
            </Typography>
        </div>
      );
    });
    const allLabelsView = this.allLabels.map((label, index) => 
      <Chip key={index} label={label}/>
    );

    return (
      <Grid 
        container
        spacing={8}
        alignItems="center"
        justify="center">
        <Grid
          container 
          justify="space-between">
          {nutrientsView}
        </Grid>
        <Grid
          container
          alignItems="center"
          justify="center">
          {allLabelsView}
        </Grid>
      </Grid>
    );
  }
}
export default IngredientNutrients;