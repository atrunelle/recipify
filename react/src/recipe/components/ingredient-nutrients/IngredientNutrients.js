import React, { Component } from "react";
import recipeService from "../../../core/recipe.service";

class IngredientNutrients extends Component {

  get allLabels () {
    const dietsLabels = recipeService.getDietLabels(this.props.nutrients.dietLabels);
    const healthLabels = recipeService.getHealthLabels(this.props.nutrients.healthLabels);
    
    return dietsLabels.concat(healthLabels);
  }
  
  get macroNutrientsData () {
    return recipeService.getMacroNutrientsList(this.props.nutrients.totalNutrients);
  }

  render() {
    const nutrientsView = this.macroNutrientsData.map((nutrient) => 
      <div>
          <h3>{ nutrient.label }</h3>
          <p>{ nutrient.quantity } { nutrient.unit }</p>
      </div>
    )
    const allLabelsView = this.allLabels.map((label) => <li>{label}</li>);

    return (
      <div>
        <div className="u-text-center">
          {nutrientsView}
        </div>
        <ul>
          {allLabelsView}
        </ul>
      </div>
    );
  }
}
export default IngredientNutrients;