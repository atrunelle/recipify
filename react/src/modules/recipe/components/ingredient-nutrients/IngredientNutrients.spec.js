import React from 'react';
import { shallow } from 'enzyme';
import IngredientNutrients from './IngredientNutrients';

describe('Component: IngredientNutrients', () => {
  it('should not render if nutrients prop is not passed', () => {
    const component = shallow(
      <IngredientNutrients/>);
    expect(component.html()).toBeNull();
  });

  it('should match snapshot', () => {
    const mockNutrients = {
      totalNutrients: {
        FAT: {
          label:'Fat',
          quantity: 13.4,
          unit: 'gr',
        },
      },
      dietLabels: ['LOW_FAT', 'LOW_CARB'],
      healthLabels: ['VEGETARIAN', 'VEGAN'],
    };
    
    const component = shallow(<IngredientNutrients nutrients={mockNutrients}/>);
    expect(component.html()).toMatchSnapshot();
  });
});