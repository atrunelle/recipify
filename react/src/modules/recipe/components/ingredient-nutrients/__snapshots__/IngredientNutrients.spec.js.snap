// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IngredientNutrients should match snapshot 1`] = `"<div class=\\"MuiGrid-container-1 MuiGrid-spacing-xs-8-22 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18\\"><div class=\\"MuiGrid-container-1 MuiGrid-justify-xs-space-between-20\\"><div><h3 class=\\"MuiTypography-root-92 MuiTypography-body1-101 MuiTypography-gutterBottom-109\\">Fat</h3><p class=\\"MuiTypography-root-92 MuiTypography-body1-101 MuiTypography-gutterBottom-109 MuiTypography-paragraph-110\\">13 gr</p></div></div><div class=\\"MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18\\"><div role=\\"button\\" class=\\"MuiChip-root-116\\" tabindex=\\"-1\\"><span class=\\"MuiChip-label-121\\">Low fat</span></div><div role=\\"button\\" class=\\"MuiChip-root-116\\" tabindex=\\"-1\\"><span class=\\"MuiChip-label-121\\">Low carb</span></div><div role=\\"button\\" class=\\"MuiChip-root-116\\" tabindex=\\"-1\\"><span class=\\"MuiChip-label-121\\">Vegetarian</span></div><div role=\\"button\\" class=\\"MuiChip-root-116\\" tabindex=\\"-1\\"><span class=\\"MuiChip-label-121\\">Vegan</span></div></div></div>"`;
